### ISSUE TITLE
Miscompilation of a `:general` form if binding in Evil's insert mode

### ISSUE THREAD

**Issue by VojtechStep** [noctuid/general.el#491]
I prefer not quoting the arguments for `general-def`, including in the `:general` section of a use-package clause. This usually results in expressions like the following:
```elisp
:general
((motion normal) global
 "C-;" #'comment-dwim)
```
```elisp
:general
(normal pdf-view-mode-map
 "r" #'image-rotate)
```
which expand to
`(general-def (motion normal) global "C-;" #'comment-dwim)`
and 
`(general-def normal pdf-view-mode-map "r" #'image-rotate)`
respectively.

However, using the same style fails for binding keys in insert mode: the snippet
```elisp
:general
(insert global
 "C-v" #'yank)
```
expands to
`(insert global "C-v" #'yank)`.

This is because the handler of the `:general` keyword checks if the first argument is callable, and if so leaves it be, to let the user call other definers instead of `general-def`.

Would it be reasonable to special-case `insert` being the first element of the list? If not, feel free to close this issue and leave it as documentation for other poor souls hitting `Error: Wrong type argument: char-or-string-p, (keymap ...)`.

**Comment by noctuid** [comment]
Yeah, the basic workaround would be to handle `insert` specially or more generally look at all keymap and state aliases.  I'm okay with that, but maybe the current check should be completely changed. It won't work in other cases too (the key definer function might not yet be defined when the check happens). Maybe there needs to be a specific list of definers to check against that can be automatically populated by `general-create-definer` at expansion time. That would probably be more inconvenient if someone was using custom definers not created with `general-create-definer` though. I'll have to think about the best way to handle this. Probably I'll just check against aliases first and then add a note to the readme that custom definers need to be available at expansion time with an example for a compiled init.

**Comment by VojtechStep** [comment]
Maybe we can use symbol properties? So that `general-create-definer` automatically sets the property on the definer symbol, and anyone using an unsupported setup can do so manually.

**Comment by noctuid** [comment]
It might cause some issues for people when updating general.el (maybe the new behavior should be opt-in for a while), but I think that's the correct way to handle it.
